AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -122,6 +124,14 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
                     self.respond(to: newState)
                 }
         }
+
+        self.outgoingMutationQueueSink = self.outgoingMutationQueue.publisher.sink { [weak self] mutationEvent in
+            guard let self = self else {
+                return
+            }
```

Comment:
are we sure we need to do [weak self] here? (I don't think we do?)

Reply:
i was following the same pattern in the few lines above, so i'm not sure
```
self.stateMachineSink = self.stateMachine
            .$state
            .sink { [weak self] newState in
                guard let self = self else {
                    return
                }
                self.log.verbose("New state: \(newState)")
                self.workQueue.async {
                    self.respond(to: newState)
                }
        }
```

should be safe to remove since these instances are long lived throughout a lifecycle of an app, and my understanding of using [weak self] is to prevent retain cycle when one of the two instances are being deinit, but neither of them should be deinitialized and if they are, probably both and all of DataStore would be torn down

Reply:
removed weak self here